_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,165 @@
+
+In **TeamCity** we define **artifacts** by defining paths to files from the working directory (which is created for each run of a configuration). The working directory is combination of files downloaded from a version control system and files generated during the execution of build steps. We will define these elements during configuring a version control settings and build steps. 
+
+We need two artifacts. The first one is `Dockerfile`. We already prepared this file in our source code and it will be stored in `docker` directory.
+The next file is a ***tar*** file which will be generated by our **Gradle** build. It contains a script which allows to execute our code and all required libraries. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_config.png" alt="Build Configuration" title="Build Configuration" class="img img-responsive style-screengrab">
+
+Now we are ready to instruct our build configuration how to download our source code. We use GitHub as our code repository.
+We just have to choose type as *Git* and provide the URL to our application.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_vcs.png" alt="Version Control Configuration" title="Version Control Configuration" class="img img-responsive style-screengrab">
+
+We have the source code, so we can describe what to do with this code. As I mentioned before we will **test** our application and generate an **artifact** with executable content.   
+
+**TeamCity** has predefined runners for different build tools. In this case we are going to use the **Gradle** runner to **test** and **build**. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_step.png" alt="Gradle Step" title="Gradle Step" class="img img-responsive style-screengrab">
+
+Now we can run our build. As result of we should see our **artifacts** in *"Artifact"* tab.
+ 
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/build_result.png" alt="Build Result" title="Build Result" class="img img-responsive style-screengrab">
+
+## Release
+
+Our code is no longer needed. We have all we need to build the docker image. Now we have to create our image and **release** with the right version. To simplify our example we are going to use the current build number to define an image version.
+Next we will generate a file with this version. This file give us the possibility to pass information about the version to the next steps.  
+
+First, let's take a closer look at the `Dockerfile`. We copy all libraries and scripts from `simple_application.tar` to the image. This ***tar*** file already contains all required content, so all we have to do is to use `ADD`. It will automatically *untar* all files in the image. Next we expose the port of our HTTP API and we define how we launch our application by adding an `ENTRYPOINT` command.  
```

Comment:
Are you trying to say "We copy `simple_application.tar`(which contains all the all libraries and scripts) to the image, by using ADD." ?